velocity/src/main/java/me/confuser/banmanager/velocity/BMVelocityPlugin.java

```diff
@@ -0,0 +1,251 @@
+        YamlConfiguration defConfig = YamlConfiguration.loadConfiguration(defConfigStream);
+        conf.setDefaults(defConfig);
+        conf.options().copyDefaults(true);
+        conf.save(file);
+      }
+    }
+
+    // Load plugin.yml
+    PluginInfo pluginInfo = new PluginInfo();
+    Reader defConfigStream = new InputStreamReader(getResourceAsStream("plugin.yml"), StandardCharsets.UTF_8);
+    YamlConfiguration conf = YamlConfiguration.loadConfiguration(defConfigStream);
+    ConfigurationSection commands = conf.getConfigurationSection("commands");
+
+    for (String command : commands.getKeys(false)) {
+      ConfigurationSection cmd = commands.getConfigurationSection(command);
+
+      pluginInfo.setCommand(new PluginInfo.CommandInfo(command, cmd.getString("permission"), cmd.getString("usage"), cmd.getStringList("aliases")));
+    }
+
+    return pluginInfo;
+  }
+
+  public void setupListeners() {
+    registerEvent(new JoinListener(this));
+    registerEvent(new LeaveListener(plugin));
+    registerEvent(new HookListener(plugin));
+
+    if (!plugin.getConfig().getChatPriority().equals("NONE")) {
+      registerEvent(new ChatListener(plugin));
+    }
+
+    if (plugin.getConfig().isDisplayNotificationsEnabled()) {
+      registerEvent(new BanListener(plugin));
+      registerEvent(new MuteListener(plugin));
+      registerEvent(new NoteListener(plugin));
+    }
+  }
+
+  private void registerEvent(Listener listener) {
+    server.getEventManager().register(this, listener);
+  }
+
+  public void setupRunnables() {
+    Runner syncRunner;
+
+    if (plugin.getGlobalConn() == null) {
+      syncRunner = new Runner(new BanSync(plugin), new MuteSync(plugin), new IpSync(plugin), new IpRangeSync(plugin), new ExpiresSync(plugin),
+          new WarningSync(plugin), new RollbackSync(plugin), new NameSync(plugin));
+    } else {
+      syncRunner = new Runner(new BanSync(plugin), new MuteSync(plugin), new IpSync(plugin), new IpRangeSync(plugin), new ExpiresSync(plugin),
+          new WarningSync(plugin), new RollbackSync(plugin), new NameSync(plugin),
+          new GlobalBanSync(plugin), new GlobalMuteSync(plugin), new GlobalIpSync(plugin), new GlobalNoteSync(plugin));
+    }
+
+    plugin.setSyncRunner(syncRunner);
+
+    setupAsyncRunnable(1L, syncRunner);
+
+    /*
+     * This task should be ran last with a 1L offset as it gets modified
+     * above.
+     */
+    setupAsyncRunnable((plugin.getSchedulesConfig()
+        .getSchedule("saveLastChecked")) + 1L, new SaveLastChecked(plugin));
+
+    // Purge
+    plugin.getScheduler().runAsync(new Purge(plugin));
+  }
+
+  private void setupAsyncRunnable(long length, Runnable runnable) {
+    if (length <= 0) return;
+
+    server.getScheduler().buildTask(this, runnable).delay(length, TimeUnit.SECONDS).schedule();
+  }
+  @SneakyThrows
+  private InputStream getResourceAsStream(String resource) {
+
+      Class<?> cls = Class.forName("me.confuser.banmanager.velocity.BMVelocityPlugin");
```

Comment:
```suggestion
      Class<?> cls = getClass();
```
If you need this to be static you could also use BMVelocityPlugin.class